api/nodes/views.py

```diff
@@ -1385,23 +1389,24 @@ class NodeLinksDetail(JSONAPIBaseView, generics.RetrieveDestroyAPIView, NodeMixi
     serializer_class = NodeLinksSerializer
     view_category = 'nodes'
     view_name = 'node-pointer-detail'
+    node_link_lookup_url_kwarg = 'node_link_id'
 
     # overrides RetrieveAPIView
     def get_object(self):
-        node_link_lookup_url_kwarg = 'node_link_id'
         node_link = get_object_or_error(
             Pointer,
-            self.kwargs[node_link_lookup_url_kwarg],
+            self.kwargs[self.node_link_lookup_url_kwarg],
             'node link'
         )
-        # May raise a permission denied
         self.check_object_permissions(self.request, node_link)
         return node_link
 
     # overrides DestroyAPIView
     def perform_destroy(self, instance):
         auth = get_user_auth(self.request)
         node = self.get_node()
+        if not node.can_edit(auth):
```

Comment:
This looks like something that was in the permissions classes you have removed (isAuthenticatedOrReadOnly, readOnlyIfRegistration, etc). Anything in this PR that is related to denying permission to do an action should almost certainly be inside a permissions class, and there hopefully shouldn't be a lot of removal of existing permission classes. Please do a pass and see how much you can put back into permissions classes now that you have a better idea of how they work.